core/config.go

```diff
@@ -0,0 +1,345 @@
+			}
+		}
+		// get the config and set it on the topic
+		topcfg, err := cfg.Riak.Service.GetTopicConfigMap(tName)
+		if err != nil {
+			// If the config object isn't found - no big deal
+			// Only config presently is a list of queues, the object will
+			// be created when the first queue is set
+			// This is a NOOP, but left in for documentation purposes
+		}
+		t.Config = topcfg
+		cfg.Topics.KnownTopics[tName] = t
+	}
+
+	// Go through the old list of topics
+	for _, topic := range oldSet {
+		tName := string(topic)
+		if _, ok := topicsToKeep[tName]; !ok {
+			// It wasn't in the old list, evict it
+			delete(cfg.Topics.KnownTopics, tName)
+		}
+	}
+
+	// Now, do the above but for the queues. Added effect - any queue removed
+	// must also be removed from a topic
+
+	qcfg, err := cfg.Riak.Service.GetQueuesConfigMap()
+	if err != nil {
+		if err != nil {
+			if err == ErrConfigMapNotFound {
+				qcfg, err = cfg.Riak.Service.CreateQueuesConfigMap()
+				if err != nil {
+					return err
+				}
+			}
+			return err
+		}
+	}
+
+	cfg.Queues.configLock.Lock()
+	defer cfg.Queues.configLock.Unlock()
+	if oldSet, ok = qcfg.Sets["queues"]; !ok {
+		// There were no known topics ?
+		// No known topics should not prevent the entire sync
+		// just the topic portion
+		log.Println(ErrNoKnownQueues)
+		return ErrNoKnownQueues
+	}
+
+	if len(oldSet) == 0 {
+		log.Println(ErrNoKnownQueues)
+		return ErrNoKnownQueues
+	}
+
+	cfg.Queues.Config = qcfg
+	if newSet, ok = cfg.Queues.Config.Sets["queues"]; !ok {
+		log.Println(ErrNoKnownQueues)
+		return ErrNoKnownQueues
+	}
+
+	queuesToRemove := make([]string, 0)
+	queuesToKeep := make(map[string]bool)
+
+	for _, queue := range newSet {
+		qName := string(queue)
+		// Record this so we know who to evict
+		queuesToKeep[qName] = true
+		// Add or Update the topic to the known set
+		var q *Queue
+		if q, ok = cfg.Queues.KnownQueues[qName]; !ok {
+			// TODO see node in LoadQueuesFromRiak about the need For
+			// and independent LoadFromRiak method
+			q, err = LoadQueueFromRiak(cfg.Riak.Service, qName)
+			if err != nil {
+				return err
+			}
+		}
+		cfg.Queues.KnownQueues[qName] = q
+	}
+
+	// Go through the old list of queues
+	for _, queue := range oldSet {
+		qName := string(queue)
+		if _, ok := queuesToKeep[qName]; !ok {
+			// It wasn't in the old list, evict it
+			delete(cfg.Queues.KnownQueues, qName)
+			queuesToRemove = append(queuesToRemove, qName)
+		}
+	}
+
+	// Go through any evicted queues, and remove them from topic subscriptions
+	// 3 loops... there has to be a better way to manage all of this
+	for _, queue := range queuesToRemove {
+		for tName, topic := range cfg.Topics.KnownTopics {
+			for _, q := range topic.Config.Sets["queues"] {
+				qName := string(q)
+				if qName == queue {
+					_, err := cfg.Riak.Service.UpdateTopicSubscription(tName, qName, false)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	log.Println("Syncing Done")
+	return nil
+}
```

Comment:
The reason why I pulled this all the way up to config, was because it was the only thing that had access to EVERYthing, where I could remove queues from the Queues collection as well as remove them from Topics by way of un-subscribe logic and removing it from the individual topic in memory at the same time.

This, and the RiakService, are a bit god-class-y, and I don't love it. But I also don't see a majorly better approach atm either.